packages/sdk/src/auth/token-provider.ts

```diff
@@ -74,8 +78,9 @@ export class CachedTokenProvider extends TokenProvider {
     if (existingDelegatePromise) {
       return existingDelegatePromise;
     }
-    const delegatePromise = this.delegate.get().then((it) => {
+    const delegatePromise = this.delegate.get().then(async (it) => {
```

Comment:
Is this a correct way of using promise.then?

Reply:
from what i understand, this is not prohibited. although it is generally recommended to avoid mixing the Promise.then pattern w/ async/await, there are exceptions (like in our case) when utilizing a promises cache (since this method must return a promise). so, we should definitely be careful for any unintended consequences, but it seems to work okay in this scenario.